contracts/REXMarket.sol

```diff
@@ -667,18 +667,6 @@ abstract contract REXMarket is Ownable, SuperAppBase, Initializable {
         require(msg.sender == address(host), "!host");
     }
 
-    /// @dev Accept only input token for CFA, output and subsidy tokens for IDA
-    function _onlyExpected(ISuperToken _superToken, address _agreementClass)
-        internal
-        view
-    {
-        if (_isCFAv1(_agreementClass)) {
-            require(_isInputToken(_superToken), "!inputAccepted");
-        } else if (_isIDAv1(_agreementClass)) {
-            require(_isOutputToken(_superToken), "!outputAccepted");
-        }
-    }
-
```

Comment:
This was the main source of the issue, using this general method which allows IDAs, this was replaced in favor of a statement that just checks that only CFAs trigger the callbacks